app/models/inventory.rb

```diff
@@ -1,9 +1,79 @@
 class Inventory < ApplicationRecord
   belongs_to :model, -> { includes :brand }
-  has_many_attached :images
+  has_many_attached :images, dependent: :destroy
   enum status: {
     active: 0,
     sold: 1,
     inactive: 2
-  }
+  }, _prefix: true
+
+  before_save :downcase_color
+
+  validates_associated :model
+  validates :memory_size, :manufactoring_year, :os_version, :color, :price, presence: true
+
+  validates :memory_size, numericality: { only_integer: true, greater_than: 0 }
+  validates :manufactoring_year, numericality: { only_integer: true, greater_than: 0,
+                                                 less_than_or_equal_to: Time.zone.now.year }
+  validates :price, numericality: { greater_than_or_equal_to: 0 }
+
+  # Search / Filter functions
+  scope :by_id, -> (id) { where id: id }
+  scope :by_model_id, -> (model_id) { where model_id: model_id }
+  scope :by_manufactoring_year_lb, -> (year) { where "manufactoring_year >= ?", year }
+  scope :by_manufactoring_year_ub, -> (year) { where "manufactoring_year <= ?", year }
+  scope :by_status, -> (status) { where status: status }
+  scope :by_os_version, -> (os) { where "os_version LIKE ?", "%" + os + "%" }
+  scope :by_color, -> (color) { where "color LIKE ?", "%" + color.downcase + "%" }
+
```

Comment:
FYI, we could use this gem to make searching is simple.
https://github.com/activerecord-hackery/ransack